data_sources/base.py

```diff
@@ -24,3 +25,7 @@ def __detect_encoding(file: BinaryIO) -> str:
                 break
         detector.close()
         return detector.result['encoding']
+    
+    @staticmethod
+    def _reg_num_is_valid(reg_num: str) -> bool:
```

Comment:
I'm not sure if it's a good idea to implement that in the data source. Data source is more about reading specific data format, not processing or filtering it.